sklearn/decomposition/dict_learning.py

```diff
@@ -246,7 +252,8 @@ def sparse_encode(X, dictionary, gram=None, cov=None, algorithm='lasso_lars',
                               algorithm=algorithm,
                               regularization=regularization, copy_cov=copy_cov,
                               init=init,
-                              max_iter=max_iter)
+                              max_iter=max_iter,
+                              check_input=check_input)
```

Comment:
That's write, it avoids multiple checking when directly calling `sparse_encode`. Done